core/src/main/java/hudson/PluginManager.java

```diff
@@ -669,9 +678,21 @@ public void dynamicLoad(File arc) throws IOException, InterruptedException, Rest
             p = strategy.createPluginWrapper(arc);
             sn = p.getShortName();
         }
-        if (getPlugin(sn)!=null)
-            throw new RestartRequiredException(Messages._PluginManager_PluginIsAlreadyInstalled_RestartRequired(sn));
-
+        PluginWrapper pw = getPlugin(sn);
+        if (pw!=null) {
```

Comment:
Agreed, and I hope to see something that enforces these trivial rules so we don't ever see discussion in PRs again :)